R/regression.R

```diff
@@ -1617,14 +1634,38 @@ removeMissingVariables <- function(data, formula, formula.with.interaction,
 # Processes the unstacked data and stacks it.
 # If the stacking is successful, the interaction, subset, weights are also updated
 # to be the appropriate size
-processAndStackData <- function(unstacked.data, formula, interaction, subset, weights)
+processAndStackData <- function(unstacked.data, formula, interaction, subset, weights, missing)
 {
     checkDataFeasibleForStacking(unstacked.data)
     unstacked.data <- removeDataReduction(unstacked.data)
     validated.unstacked.output <- validateDataForStacking(unstacked.data)
     unstacked.data <- validated.unstacked.output[["data"]]
     stacks <- validated.unstacked.output[["stacks"]]
-    data <- stackData(unstacked.data)
+    stacked.data <- stackData(unstacked.data)
+    n.orig.stacked.cases = nrow(stacked.data)
+
+    # Exclude missing cases. Stacking can result in
+    # many empty cases when data set is sparse. Respondent-level
+    # functions (e.g. predicted values) are not supported
+    # for stacked data, so no need to keep track of removed
+    # cases.
+    rm.missing <- rep(FALSE, n.orig.stacked.cases)
```

Comment:
`logical(n.orig.stacked.cases)` is slightly faster (a micro optimisation) but slightly less readable. No need to change just an FYI